packages/shared/friends/sagas.ts

```diff
@@ -2487,3 +2525,85 @@ function buildFriendRequestErrorResponse(error: FriendshipErrorCode) {
 function buildFriendRequestReply<T>(reply: NonNullable<T>) {
   return { reply, error: undefined }
 }
+
+/**
+ * Check whether the user has reached the max number allowed of sent requests to a given user.
+ * @param userId - the user to check the number of requests for.
+ * @returns true if the user has reached the max number of sent requests, false otherwise
+ */
+function reachedMaxNumberOfRequests(userId: string) {
+  // Get number friend requests sent in the current session to the given user
+  const sentRequests = getNumberOfFriendRequests(store.getState())
+  const number = sentRequests.get(userId) ?? 0
+
+  // Get the maximum number of requests allowed
+  const maxNumber = DEFAULT_MAX_NUMBER_OF_REQUESTS
+
+  // Check if the current number of requests is less than the maximum allowed
+  return number < maxNumber ?? false
+}
+
+/**
+ * Check whether there is a remaining cooldown time to send a friend request to a given user.
+ * @param userId - the user to check the cooldown time for.
+ * @returns true if there is a remaining cooldown time and it hasn't expired yet, false otherwise
+ */
+function hasRemainingCooldown(userId: string) {
+  const currentTime = Date.now()
+
+  // Get the remaining cooldown time for the given user
+  const coolDownTimer = getCoolDownOfFriendRequests(store.getState())
+  const remainingCooldownTime = coolDownTimer.get(userId)
+
+  // If there is a remaining cooldown time and it hasn't expired yet, return false
+  return (remainingCooldownTime && currentTime < remainingCooldownTime) ?? false
+}
+
+/**
+ * Filters and processes (update their friendship status as rejected) friend requests from blocked users.
+ * @param fromFriendRequests - an array of from friend requests.
+ */
+function handleBlockedUsers(fromFriendRequests: FriendRequest[]) {
+  // Get the ids of users who have been blocked
+  const blockedIds = fromFriendRequests.filter((fromFriendRequest) => isBlocked(fromFriendRequest.userId))
+
+  // For each blocked user, update their friendship status as rejected
+  const promises = blockedIds.map(async (fromFriendRequest) => {
+    return handleBlockedUser(fromFriendRequest.userId, fromFriendRequest.message)
+  })
+
+  // Wait for all Promises to resolve
+  return Promise.all(promises)
+}
+
+/**
+ * Processes (update their friendship status as rejected) a friend request from a blocked user.
+ * @param id - the id of the user whose friend request is being processed.
+ */
+async function handleBlockedUser(id: string, message?: string) {
+  // Update their friendship status as rejected
+  await UpdateFriendshipAsPromise(FriendshipAction.REJECTED, id.toLowerCase(), false, message)
```

Comment:
Return promise result.
```suggestion
  return await UpdateFriendshipAsPromise(FriendshipAction.REJECTED, id.toLowerCase(), false, message)
```